src/bicep/mlz.bicep

```
@@ -252,6 +252,62 @@ module sharedServicesVirtualNetworkPeering './modules/spokeNetworkPeering.bicep'
   }
 }
 
+module hubPolicyAssignment './modules/policyassignment.bicep' = {
+  name: 'policyAssignement'
+  scope: resourceGroup(hubSubscriptionId, hubResourceGroupName)
+  dependsOn: [
+    hubResourceGroup
+    logAnalyticsWorkspace
+  ]
+  params: {
+    builtInAssignment: policy
+    logAnalyticsWorkspaceName: logAnalyticsWorkspace.outputs.name
+    workspaceResourceGroupName: operationsResourceGroupName
+  }
+}
+
+module operationsPolicyAssignment './modules/policyassignment.bicep' = {
+  name: 'policyAssignement'
+  scope: resourceGroup(operationsSubscriptionId, operationsResourceGroupName)
+  dependsOn: [
+    operationsResourceGroup
+    logAnalyticsWorkspace
+  ]
+  params: {
+    builtInAssignment: policy
+    logAnalyticsWorkspaceName: logAnalyticsWorkspace.outputs.name
+    workspaceResourceGroupName: operationsResourceGroupName
+  }
+}
```

Want to say that for all these policy assignment modules:

1. the module names need to be unique at this scope at deployment-time, otherwise the first one will start and the subsequent deployments will run into naming conflicts
1. we should be able to use the symbolic resource of the relevant Resource Group and the Log Analytics Workspace allowing Bicep figure out the implied dependency and remove the explicit `dependsOn`:

```suggestion
module operationsPolicyAssignment './modules/policyAssignment.bicep' = {
  name: '${operationsResourceGroupName}-policyAssignment'
  scope: resourceGroup(operationsSubscriptionId, operationsResourceGroupName)
  params: {
    builtInAssignment: policy
    logAnalyticsWorkspaceName: logAnalyticsWorkspace.outputs.name
    workspaceResourceGroupName: operationsResourceGroup.name
  }
}
```

---

Agreed, although it was not honoring the implied depend dependency, so I made it explicate to avoid the error message.

---

> Agreed, although it was not honoring the implied depend dependency, so I made it explicate to avoid the error message.

Ah, is that so? Figured there was a reason for that. Disregard the dependsOn then.